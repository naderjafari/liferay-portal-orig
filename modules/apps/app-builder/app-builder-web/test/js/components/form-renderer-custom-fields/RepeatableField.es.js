/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {cleanup, render} from '@testing-library/react';
import React, {createContext} from 'react';

import RepeatableField from '../../../../src/main/resources/META-INF/resources/js/components/form-renderer-custom-fields/RepeatableField.es';

const MyContext = createContext();

const RepeatableFieldWrapper = ({children, state}) => (
	<MyContext.Provider value={[state]}>
		<RepeatableField AppContext={MyContext}>{children}</RepeatableField>
	</MyContext.Provider>
);

describe('RepeatableField', () => {
	afterEach(() => {
		cleanup();
	});

	it('renders with fieldset focusedField', () => {
		const state = {
			focusedCustomObjectField: {fieldType: 'fieldset'},
			focusedField: {type: 'fieldset'},
		};

		const {asFragment, queryByText} = render(
			<RepeatableFieldWrapper state={state}>
				{() => <b>Original Component</b>}
			</RepeatableFieldWrapper>
		);

		expect(queryByText('Original Component')).toBeNull();

		expect(asFragment()).toMatchSnapshot();
	});

	it('renders with text focusedField', () => {
		const state = {
			focusedCustomObjectField: {fieldType: 'text'},
			focusedField: {type: 'text'},
		};
		const {asFragment, queryByText} = render(
			<RepeatableFieldWrapper state={state}>
				{() => <b>Original Component</b>}
			</RepeatableFieldWrapper>
		);

		expect(queryByText('Original Component')).toBeTruthy();

		expect(asFragment()).toMatchSnapshot();
	});
});
