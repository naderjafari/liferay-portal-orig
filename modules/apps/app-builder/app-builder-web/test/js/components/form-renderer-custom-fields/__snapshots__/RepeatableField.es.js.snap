// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepeatableField renders with fieldset focusedField 1`] = `<DocumentFragment />`;

exports[`RepeatableField renders with text focusedField 1`] = `
<DocumentFragment>
  <b>
    Original Component
  </b>
</DocumentFragment>
`;
